File: app/src/main/java/com/example/capstone_donkki/Ingredient_Adapter.java
package com.example.capstone_donkki;


import android.content.ClipData;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NonNls;

import java.util.List;

public class Ingredient_Adapter extends RecyclerView.Adapter<Ingredient_Adapter.ItemViewHolder> {

    private List<IngredientData> IngredientList;

    public Ingredient_Adapter(List<IngredientData> IngredientList) {
        this.IngredientList = IngredientList;
    }

    public static class ItemViewHolder extends RecyclerView.ViewHolder {
        public ImageView IngredientImage;
        public TextView IngredientName, expiration;

        public ItemViewHolder(View itemView) {
            super(itemView);
            IngredientImage = itemView.findViewById(R.id.iv_ingredient);
            IngredientName = itemView.findViewById(R.id.tv_ingredient);
            expiration = itemView.findViewById(R.id.tv_expiration);


        }
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View ItemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.frag_main, viewGroup, false);

        return new ItemViewHolder(ItemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder viewHolder, int i) {
        IngredientData ingredient = IngredientList.get(i);
        viewHolder.expiration.setText(ingredient.getExpiration());
        viewHolder.IngredientName.setText(ingredient.getIngredientName());
        viewHolder.IngredientImage.setImageResource(ingredient.getIngredientImage());


    }

    @Override
    public int getItemCount() {
        return IngredientList.size();
    }

}

File: app/src/main/java/com/example/capstone_donkki/Ingredient_Adapter.kt
package com.example.capstone_donkki

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

//리사이클러 뷰 어뎁터 설정
class Ingredient_Adapter(private val ingredientList: ArrayList<IngredientData>) :
    RecyclerView.Adapter<Ingredient_Adapter.CustomViewHolder>() {
    //새로운 뷰 생성
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
        //화면 연결
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.ref_ingredient_list, parent, false)

        return CustomViewHolder(view).apply {
            itemView.setOnClickListener {
                //리사이클러 뷰 내부 아이템 클릭 이벤트


            }
        }
    }

    override fun getItemCount(): Int {
        return ingredientList.size
    }

    override fun onBindViewHolder(holder: Ingredient_Adapter.CustomViewHolder, position: Int) {
        //뷰에 아이템 리스트를 바인딩
        holder.ingredientImage.setImageResource(ingredientList.get(position).ingredientImage)
        holder.ingredientText.text = ingredientList.get(position).ingredientName
        holder.expiration.text = ingredientList.get(position).expiration

    }

    class CustomViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val ingredientImage = itemView.findViewById<ImageView>(R.id.iv_ingredient)    //식재료 이미지
        val ingredientText = itemView.findViewById<TextView>(R.id.tv_ingredient)      //식재료명
        val expiration = itemView.findViewById<TextView>(R.id.tv_expiration)          //소비기한

    }
}